hoymiles_mqtt/__main__.py

```diff
@@ -16,6 +16,24 @@
 DEFAULT_QUERY_PERIOD_SEC = 60
 DEFAULT_MODBUS_UNIT_ID = 1
 
+logger = logging.getLogger(__name__)
+
+
+def _setup_logger(options: configargparse.Namespace) -> None:
+    log_level_parameter = options.log_level
+    if log_level_parameter.upper() not in logging._nameToLevel:
```

Comment:
Would it be possible to use https://docs.python.org/3.12/library/logging.html#logging.getLevelName instead of the private interface?

Reply:
What about leveraging argaprse to validate the input? See "choices" below:

    cfg_parser.add(
        '--log-level',
        required=False,
        type=str,
        default='WARNING',
        choices=['DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL'],
        env_var='LOG_LEVEL',
        help="Python logger log level.",
    )

Reply:
Changed